packages/contracts/contracts/optimistic-ethereum/ovm/ExecutionManager.sol

```diff
@@ -545,6 +533,8 @@ contract ExecutionManager {
      * returndata: [newOvmContractAddress (as bytes32)] -- will be all 0s if this create failed.
      */
     function ovmCREATE() public {
+        StateManager stateManager = resolveStateManager();
```

Comment:
Oh one thing that I was thinking -- because resolveStateManager() slightly increases our gas costs & we use it soooo much, it could be good in this case to actually store an internal variable in the EM which just is set to `resolveStateManager()` at initialization.

Reply:
Could say the same for a bunch of the `resolve...` calls in the EM specifically. It's just such a hot spot wrt code that any gas saving is a blessing. In fact we should probably create a ticket for going through the EM and reducing gas costs where possible

Reply:
Hmmm yes although this would also limit the ability to change `StateManager` via the `AddressResolver` (unless we simply want to use the `setStateManager` function here as an exception).

Reply:
Hmm yeah I feel like it might be worth it to have the exception in this case, just because the overhead will be at least like ~2k gas per tx otherwise